test-engine-core/test_engine_core/plugins/plugins_manager.py

```diff
@@ -422,29 +423,23 @@ def _sort_plugin(plugin_tuple: Tuple[str, IPlugin]) -> int:
         plugin_type = plugin_tuple[1].Plugin.get_plugin_type()
 
         if plugin_type is PluginType.DATA:
-            return PluginManager._data_priority_list.index(
-                plugin_tuple[1].Plugin.get_data_plugin_type()
-            )
-
+            prio_type = plugin_tuple[1].Plugin.get_data_plugin_type()
+            priority = PluginManager._data_priority_list
         elif plugin_type is PluginType.MODEL:
-            return PluginManager._model_priority_list.index(
-                plugin_tuple[1].Plugin.get_model_plugin_type()
-            )
-
+            prio_type = plugin_tuple[1].Plugin.get_model_plugin_type()
+            priority = PluginManager._model_priority_list
         elif plugin_type is PluginType.PIPELINE:
-            return PluginManager._pipeline_priority_list.index(
-                plugin_tuple[1].Plugin.get_pipeline_plugin_type()
-            )
-
+            prio_type = plugin_tuple[1].Plugin.get_pipeline_plugin_type()
+            priority = PluginManager._pipeline_priority_list
         elif plugin_type is PluginType.SERIALIZER:
-            return PluginManager._serializer_priority_list.index(
-                plugin_tuple[1].Plugin.get_serializer_plugin_type()
-            )
-
+            prio_type = plugin_tuple[1].Plugin.get_serializer_plugin_type()
+            priority = PluginManager._serializer_priority_list
         else:
-            return list(
-                PluginManager._plugins[PluginType.ALGORITHM.name].values()
-            ).index(plugin_tuple[1])
+            prio_type = plugin_tuple[1]
+            priority = PluginManager._plugins[PluginType.ALGORITHM.name].values()
+
+        return priority.index(prio_type) if prio_type in priority else len(priority)
```

Comment:
Curious what you think, took me a while to find why my new model wouldn't be incorporated. This change allows for adding models without specifying their priority (putting them last).